src/FlashtestationRegistry.sol

```diff
@@ -147,4 +153,51 @@ contract FlashtestationRegistry {
     function isValidWorkload(WorkloadId workloadId, address teeAddress) public view returns (bool) {
         return WorkloadId.unwrap(registeredTEEs[teeAddress].workloadId) == WorkloadId.unwrap(workloadId);
     }
+
+    /**
+     * @notice Re-verifies the attestation of a TEE
+     * @param teeAddress The TEE-controlled address to re-verify
+     * @dev This is a costly operation (5 million gas) and should be used sparingly.
+     * @dev Will always revert except if the attestation is no invalid. This is to prevent
+     * a user needlessly calling this function and for a no-op to occur
+     * @dev This function exists to handle an important security requirement: occasionally Intel
+     * will release a new set of DCAP Endorsements for a particular TEE setup (for instance if a
+     * TDX vulnerability was discovered), which invalidates all prior quotes generated by that TEE.
+     * By invalidates we mean that the outputs generated by the TEE-controlled address associated
+     * with these invalid quotes are no longer secure and cannot be relied upon. This fact needs to be
+     * reflected onchain, so that any upstream contracts that try to call `isValidWorkload` will
+     * correctly return `false` for the TEE-controlled addresses associated with these invalid quotes.
+     * This is a security requirement to ensure that no downstream contracts can be exploited by
+     * a malicious TEE that has been compromised
+     * @dev Note: this function is callable by anyone, so that offchain monitoring services can
+     * quickly mark TEEs as invalid
+     */
+    function reverifyAttestation(address teeAddress) external {
+        // check to make sure it even makes sense to re-verify the TEE-controlled address
+        // if the TEE-controlled address is not registered with the FlashtestationRegistry,
+        // it doesn't make sense to re-verify the attestation
+        RegisteredTEE memory registeredTEE = registeredTEEs[teeAddress];
+        if (WorkloadId.unwrap(registeredTEE.workloadId) == 0) {
+            revert TEEServiceNotRegistered(teeAddress);
+        }
+
+        if (!registeredTEE.isValid) {
+            revert TEEServiceAlreadyInvalid(teeAddress);
+        }
+
+        // now we re-verify the attestation, and invalidate the TEE if it's no longer valid.
+        // This will only happen if the DCAP Endorsements associated with the TEE's quote
+        // have been updated
+        (bool success,) = attestationContract.verifyAndAttestOnChain(registeredTEE.rawQuote);
+        if (success) {
+            // if the attestation is still valid, then this function call is a no-op except for
```

Comment:
note: i went back and forth with whether to revert if the call to `reverifyAttestation` would be a no-op and only waste gas, or let it go through and return `true` to the user but waste their gas. I went with reverting because that is the common way to write smart contracts (try to prevent the user from doing no-ops), and we have 2 more reverts up above that the caller will need to check for anyway. If you have strong thoughts on this, for instance you are adamant that the caller should be able to call `reverifyAttestation` and get `true` or `false` back, I'd love to hear it.

Reply:
I like your approach the most